frame/vesting/src/lib.rs

```diff
@@ -233,10 +236,14 @@ decl_module! {
 		/// Emits `VestingCreated`.
 		///
 		/// # <weight>
-		/// - Creates a new storage entry, but is protected by a minimum transfer
-		///	   amount needed to succeed.
+		/// - `O(1)`.
+		/// - DbWeight: 3 Reads, 3 Writes
+		///     - Reads: Vesting Storage, Balances Locks, Target Account, [Sender Account]
+		///     - Writes: Vesting Storage, Balances Locks, Target Account, [Sender Account]
+		/// - Benchmark: 263 µs (min square analysis)
+		/// - Assuming less than 50 locks on any user, else we may want factor in number of locks.
 		/// # </weight>
-		#[weight = 1_000_000_000]
+		#[weight = 300_000_000 + T::DbWeight::get().reads_writes(3, 3)]
```

Comment:
do we have a rule of thumb for granularity or soundings? This, accurately, should have been around 260_000_000 or sth?

Reply:
Benchmarks are not that accurate to begin with, but yeah, rounding here is mostly just an opinion, as long as it does not extremely change the value.

Reply:
Benchmarks are not that accurate to begin with, but yeah, rounding here is mostly just an opinion, as long as it does not extremely change the value.

I am also taking into consideration the variability of locks. If this is a little bit larger, it is a little bit safer overall as more locks are introduced per user.